graphqlapi/src/main/java/io/stargate/graphql/web/Server.java

```diff
@@ -0,0 +1,130 @@
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+public class Server extends Application<ApplicationConfiguration> {
+
+  private final Persistence persistence;
+  private final AuthenticationService authenticationService;
+  private final Metrics metrics;
+  private volatile org.eclipse.jetty.server.Server jettyServer;
+
+  public Server(
+      Persistence persistence, AuthenticationService authenticationService, Metrics metrics) {
+    this.persistence = persistence;
+    this.authenticationService = authenticationService;
+    this.metrics = metrics;
+  }
+
+  /**
+   * The only reason we override this is to remove the call to {@code bootstrap.registerMetrics()}.
+   *
+   * <p>JVM metrics are registered once at the top level in the health-checker module.
+   */
+  @Override
+  public void run(String... arguments) {
+    final Bootstrap<ApplicationConfiguration> bootstrap = new Bootstrap<>(this);
+    addDefaultCommands(bootstrap);
+    initialize(bootstrap);
+
+    final Cli cli = new Cli(new JarLocation(getClass()), bootstrap, System.out, System.err);
+    // only exit if there's an error running the command
+    cli.run(arguments).ifPresent(this::onFatalError);
+  }
+
+  @Override
+  public void run(
+      final ApplicationConfiguration applicationConfiguration, final Environment environment) {
+
+    GraphqlCache graphqlCache = new GraphqlCache(persistence, authenticationService);
+    environment
+        .jersey()
+        .register(
+            new AbstractBinder() {
+              @Override
+              protected void configure() {
+                bind(graphqlCache).to(GraphqlCache.class);
+              }
+            });
+    environment
+        .jersey()
+        .register(
+            new AbstractBinder() {
+              @Override
+              protected void configure() {
+                bind(FrameworkUtil.getBundle(GraphqlActivator.class)).to(Bundle.class);
+              }
+            });
+
+    environment.jersey().register(GraphqlDmlResource.class);
+    environment.jersey().register(GraphqlDdlResource.class);
+    environment.jersey().register(PlaygroundResource.class);
+
+    enableCors(environment);
+
+    environment.lifecycle().addServerLifecycleListener(server -> Server.this.jettyServer = server);
+  }
+
+  @Override
+  public void initialize(final Bootstrap<ApplicationConfiguration> bootstrap) {
+    super.initialize(bootstrap);
+    bootstrap.setConfigurationSourceProvider(new ResourceConfigurationSourceProvider());
+    bootstrap.setMetricRegistry(metrics.getRegistry("graphqlapi"));
+  }
+
+  private void enableCors(Environment environment) {
+    FilterRegistration.Dynamic filter =
+        environment.servlets().addFilter("cors", CrossOriginFilter.class);
+
+    filter.setInitParameter(
+        CrossOriginFilter.ALLOWED_METHODS_PARAM, "POST,GET,OPTIONS,PUT,DELETE,PATCH");
+    filter.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
```

Comment:
is it safe to allow all?

Reply:
We plan to make this configurable (e.g. provide a single host or list of hosts). I believe there's an issue somewhere for it.

Reply:
It is indeed very permissive.
It seems hard to come up with a good value though, because it will typically depend on the user's environment. We can't predict where their client applications will be located. Maybe we should make this configurable. @dougwettlaufer any opinion?

Reply:
Yep, as I mentioned above we'll make all of the CORs settings configurable later in another ticket.